### DISCUSSION TITLE
pandoc LaTeX > ePub <img> gets no 'alt="" ' anymore

### DISCUSSION THREAD

**Discussion by sithscho** [jgm/pandoc#9347] [General]
In former Versions every <img> in an ePub converted from LaTeX got an 'alt=""' parameter…

But in the most recent version of pandoc this is not the case anymore.

How do I tell pandoc to create an 'alt=""' parameter either…
- using the \caption{} from the figure environment
- or just writing an empty alt="" as before…

???

(iBooks requires that alt="" parameter for some reason…)

**Comment by jgm** [comment]
> (iBooks requires that alt="" parameter for some reason…)

Can you elaborate?  What happens if it's missing?

**Comment by jgm** [comment]
Here's what I'm seeing with latest pandoc:
```
% pandoc -f latex -t html
\includegraphics{img.jpeg}
<p><img src="img.jpeg" alt="image" /></p>
```
I assume that the alt is also included when the HTML is embedded in EPUB [confirmed].  So I think we need more details about what you're doing -- what the latex image syntax is, what version of pandoc.

**Comment by sithscho** [comment]
Thank you in advance.

It’s pandoc 3.1.11.1

Syntax is: 
`pandoc --from=latex --to=epub2 --epub-cover-image=Cover.jpg source.tex --output=result.epub --variable=lang:fr`

Tried it with epub3, result/problem is the same.

**Latex Source**:
```
\begin{figure}[h]
\centering
\includegraphics[width=0.99\textwidth]{../pic/image1.jpg}
\caption{Image1 Caption}
\end{figure}
```

**ePub Result**:

```
<div class="float">
<img src="../media/file6.jpg" style="width:99.0%" />
<div class="figcaption">Image1 Caption</div>
</div> 
```

Apple iBooks gives an error because of the missing alt="" detail. 
No more comments about it, just: »Error missing alt parameter in img.«


In a former Pandoc Version (one for MacOS 10.15 ‚Catalina‘ from 2019) the epub code used to look like this:

```
<div class="figure">
<img src="../media/file6.jpg" style="width:99.0%" alt="" />
<p class="caption">Image1 Caption</p>
</div> 
```
That used to work everywhere.

**Comment by jgm** [comment]
I'm seeing this in the pandoc 3.0 release notes:

> HTML writer: the alt text is no longer constructed from the caption, as was the case with implicit figures. This reduces duplication, but comes at the risk of images that are missing alt texts. Authors should take care to provide alt texts for all images. Some readers, most notably the Markdown reader with the `implicit_figures` extension, add a caption that’s identical to the image description. The writer checks for this and adds an `aria-hidden` attribute to the <figcaption> element in that case.

This was part of a huge change adding the Figure element (commit 909ced5153e2c7cefd5018c39f83231824940fb8).

@tarleb - should we just add an alt text based on the caption when alt text is missing?

**Reply by tarleb** [reply]
I think issue #4737 is relevant for that, as is #6782. At first glance it seems that the former argues against using the caption as alt text, while the latter seems to say it's ok. But I'll have to re-read those issues in detail.

Maybe @jooyoungseo has time to share their expertise on this issue?

**Comment by sithscho** [comment]
I think there is a reason, why Apple demands the alt-text. Probably this is due to some standard or definition that has to be met.

At least there should be a switch to tell the HTML writer to insert or not-insert an alt-text.

Btw: the alt-text was always empty. So the Latex-Caption > Alt-Text has not worked. It sure would be nice to have that working also…

**Comment by sithscho** [comment]
so now…?

**Reply by tarleb** [reply]
The code was modified to make sure an `alt` attribute will always be present when EPUB is generated.

**Reply by sithscho** [reply]
Thank you very much. That helps a lot!
